StackExchange.Redis/StackExchange/Redis/ResultProcessor.cs

```diff
@@ -1301,6 +1325,461 @@ protected override bool SetResultCore(PhysicalConnection connection, Message mes
+        {
+            protected override StreamConsumerInfo ParseItem(RawResult result)
+            {
+                // Note: the base class passes a single consumer from the response into this method.
+
+                // Response format:
+                // > XINFO CONSUMERS mystream mygroup
+                // 1) 1) name
+                //    2) "Alice"
+                //    3) pending
+                //    4) (integer)1
+                //    5) idle
+                //    6) (integer)9104628
+                // 2) 1) name
+                //    2) "Bob"
+                //    3) pending
+                //    4) (integer)1
+                //    5) idle
+                //    6) (integer)83841983
+
+                var arr = result.GetItems();
+
+                var consumerName = (string)arr[1].AsRedisValue();
+                var pendingCount = (int)arr[3].AsRedisValue();
+                var idleTime = (int)arr[5].AsRedisValue();
+
+                return new StreamConsumerInfo(consumerName, pendingCount, idleTime);
+            }
+        }
+
+        internal sealed class StreamGroupInfoProcessor : InterleavedStreamInfoProcessorBase<StreamGroupInfo>
+        {
+            protected override StreamGroupInfo ParseItem(RawResult result)
+            {
+                // Note: the base class passes a single item from the response into this method.
+
+                // Response format:
+                // > XINFO GROUPS mystream
+                // 1) 1) name
+                //    2) "mygroup"
+                //    3) consumers
+                //    4) (integer)2
+                //    5) pending
+                //    6) (integer)2
+                // 2) 1) name
+                //    2) "some-other-group"
+                //    3) consumers
+                //    4) (integer)1
+                //    5) pending
+                //    6) (integer)0
+
+                var arr = result.GetItems();
+
+                var groupName = (string)arr[1].AsRedisValue();
+                var consumerCount = (int)arr[3].AsRedisValue();
+                var pendingCount = (int)arr[5].AsRedisValue();
+
+                return new StreamGroupInfo(groupName, consumerCount, pendingCount);
+            }
+        }
+
+        internal abstract class InterleavedStreamInfoProcessorBase<T> : ResultProcessor<T[]>
+        {
+            protected abstract T ParseItem(RawResult result);
+
+            protected override bool SetResultCore(PhysicalConnection connection, Message message, RawResult result)
+            {
+                if (result.Type != ResultType.MultiBulk)
+                {
+                    return false;
+                }
+
+                var arr = result.GetItems();
+                var parsedItems = new T[arr.Length];
+
+                for (var i = 0; i < arr.Length; i++)
+                {
+                    parsedItems[i] = ParseItem(arr[i]);
+                }
```

Comment:
Can simplify this and other similar loops using `Array.ConverAll`, e.g.
```c#
var parsedItems = Array.ConvertAll(arr, ParseItem);
```

Reply:
Nit for @NickCraver: prefer:

```
var parsedItems = Array.ConvertAll(arr, item => ParseItem(item));
```

Unless I'm mistaken, they're still subtly different. Happy to find out they fixed the compiler, of course.